data/tests/test_models.py

```diff
@@ -37,3 +46,243 @@ def test_class_meta_ordering(self):
+        hoster = issue._meta.get_field('hoster').verbose_name
+        labels = issue._meta.get_field('labels').verbose_name
+        assignees = issue._meta.get_field(
+            'assignees').verbose_name
+        self.assertEquals(number, 'number')
+        self.assertEquals(title, 'title')
+        self.assertEquals(repo_id, 'repo id')
+        self.assertEquals(repo, 'repo')
+        self.assertEquals(author, 'author')
+        self.assertEquals(created_at, 'created at')
+        self.assertEquals(updated_at, 'updated at')
+        self.assertEquals(state, 'state')
+        self.assertEquals(hoster, 'hoster')
+        self.assertEquals(labels, 'labels')
+        self.assertEquals(assignees, 'assignees')
+
+    def test_object_name_is_title(self):
+        issue = Issue.objects.get(number=1)
+        expected_object_name = 'Test issue'
+        self.assertEquals(expected_object_name, str(issue))
+
+    def test_many_to_many_field(self):
+        issue = Issue.objects.get(number=1)
+        label = Label.objects.get(id=1)
+        assignee = Contributor.objects.get(login='sks444')
+
+        # Test issue has many to many field with label
+        self.assertEquals(issue.labels.get(pk=label.pk),
+                          label)
+
+        # Test issue has many to many field with contributor
+        self.assertEquals(issue.assignees.get(pk=assignee.pk),
+                          assignee)
+
+
+class IssueNumberModelTest(TestCase):
+
+    @classmethod
+    def setUpTestData(cls):
+        # Set up non-modified objects used by all methods
+        contributor = Contributor.objects.create(login='sks444',
+                                                 name='Shrikrishna Singh')
+        created_at = pytz.utc.localize(parse('2017-11-07T08:01:21'))
+        updated_at = pytz.utc.localize(parse('2018-03-18T00:00:11'))
+        Issue.objects.create(number=1,
+                             title='Test issue',
+                             repo_id=1,
+                             author=contributor,
+                             created_at=created_at,
+                             updated_at=updated_at,
+                             state='open',
+                             hoster='GitHub')
+        IssueNumber.objects.create(number=1, repo_id=1)
+
+    def test_field_label(self):
+        issue_number = IssueNumber.objects.get(number=1)
+        number = issue_number._meta.get_field('number').verbose_name
+        repo_id = issue_number._meta.get_field('repo_id').verbose_name
+        self.assertEquals(number, 'number')
+        self.assertEquals(repo_id, 'repo id')
+
+    def test_object_name_is_number(self):
+        issue_number = IssueNumber.objects.get(number=1)
+        expected_object_name = '1'
+        self.assertEquals(expected_object_name, str(issue_number))
+
+    def test_get_issue_for_issue_number(self):
+        issue_number = IssueNumber.objects.get(number=1)
+        issue = Issue.objects.get(number=1)
```

Comment:
need a real test of a real MR in real repo, which is merged and closed an issue

I cant see which repo this refers to